src/app/services/task.service.ts

```diff
@@ -24,10 +24,9 @@ export class TaskService {
   addTask(newTask: TaskModel): void {
     const tasks = this.taskSubject.getValue();
 
-    this.http.post(`${environment.apiEndpoint}/tasks`, newTask)
-      .subscribe(res => {
-        const task = Array.isArray(res) ? res[0] : res;
-        this.taskSubject.next([...tasks, task]);
+    this.http.post<TaskModel>(`${environment.apiEndpoint}/tasks`, newTask)
+      .subscribe((res: TaskModel) => {
```

Comment:
I would not subscribe here, as I said previously maintaining subscriptions in the service is very hard, it is much easier to subscribe using the async pipe or subscribing and unsubscribing in the components